webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -610,3 +627,14 @@ def _get_first_observation_x(obs_data: pd.DataFrame) -> Union[int, str]:
         )
 
     return caster.get(type(first_observation), lambda *args: False)(first_observation)
+
+
+def _format_index_value(axis: pd.Index, index: int) -> Union[int, datetime.date]:
+    raw_value = axis[index]
+    if isinstance(raw_value, str):
+        try:
+            datetime_value = pd.to_datetime(raw_value)
+            return datetime_value.date()
+        except (pd.errors.ParserError):
+            pass
+    return raw_value
```

Comment:
This function could be tested. It would be good to document which states we expect to have a proper value and which states the exception is to be raised.

Reply:
thanks for pointing that out - I wrote a parametrized test, and found an error (or not considered case) in the code, so that was nice ^^

have a look and see if it's alright please : )